impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -331,20 +354,30 @@ else if (searchScope == LdapSearchScope.SUBTREE) {
     }
 
     private LdapContext createDefaultLdapContext() {
-        return createLdapContext(
+        try {
+          return createLdapContext(
                 ldapIdentityStoreDefinition.url(),
                 ldapIdentityStoreDefinition.bindDn(),
                 ldapIdentityStoreDefinition.bindDnPassword());
+        } catch (Exception e) {
+          LOGGER.log(Level.SEVERE, "Error creating default ldap context " + e);
+        }
+        return null;
     }
 
-    private static LdapContext createLdapContext(String url, String bindDn, String bindCredential) {
+    private static LdapContext createLdapContext(String url, String bindDn, String bindCredential) throws Exception {
```

Comment:
This forces every place that calls `createLdapContext()` to handle exceptions. If we're careful about how we handle them, we can allow runtime exceptions to just propagate.

It does mean that we might want to separate `createDefaultLdapContext()` and `createLdapContext()` a little more, so they don't call each other. We could either move the common code into a third method that they both call, and which passes through all exceptions -- so that the two methods can handle them differently -- or we could just have each one duplicate the one line of code that's used to actually create the context; i.e.,:
```
return new InitialLdapContext(getConnectionEnvironment(url, bindDn, bindCredential), null);
```
I guess my preference would be to share the common code in a third method anyway, but the that method is completely trivial. The exception handling part is hard, but must be done differently for each "flavor" of `createLdapContext()`.